third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -1515,6 +1515,32 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         }
     }
 
+    /// Instantiate types for all all struct patterns in the block expression
+    /// This step is necessary because struct pattern may contain uninstantiated variable types
+    pub fn rewrite_block_pattern(&mut self, result_exp: Exp) -> Exp {
+        ExpData::rewrite(result_exp, &mut |e| {
+            let exp_data: ExpData = e.into();
+            if let ExpData::Block(id, pat, opt_exp, exp) = exp_data {
+                if let Struct(sid, std, patterns) = pat {
```

Comment:
Please no unqualified user of symbols from other modules. Unless there is a function local use, e.g. as in `fn foo(e: Exp) { use ExpData::*'; ... }`